powerlevel9k.zsh-theme

```diff
@@ -337,6 +337,47 @@ prompt_background_jobs() {
   fi
 }
 
+# Segment that indicates usage level of current partition.
+set_default POWERLEVEL9K_HDD_USAGE_ONLY_WARNING false
+set_default POWERLEVEL9K_HDD_USAGE_WARNING_LEVEL 90
+set_default POWERLEVEL9K_HDD_USAGE_CRITICAL_LEVEL 95
+prompt_hdd_usage() {
+  local current_state="unknown"
+  typeset -AH hdd_usage_forecolors
+  hdd_usage_forecolors=(
+    'normal'        'yellow'
+    'warning'       "$DEFAULT_COLOR"
+    'critical'      'white'
+  )
+  typeset -AH hdd_usage_backcolors
+  hdd_usage_backcolors=(
+    'normal'        $DEFAULT_COLOR
+    'warning'       'yellow'
+    'critical'      'red'
+  )
+
+  local level="${$(df -k . | sed -n '2p' | awk '{ print $5 }')%%\%}"
```

Comment:
So the problem with that structure is that the 'CRITICAL_LEVEL` conditional would never return `true`. Because if its `>=` the critical level, it's also `>=` the warning level, and the critical level check is an `else if`, it will never be true. I think that the way that I have it now minimizes the number of conditionals that are evaluated for any one state.

I'm totally open to other structures, though - I agree that the flow is a bit confusing :-/